### PR TITLE
added authentication and notes content

### PR DESCRIPTION
**Pull request by ShreyashMahajan**
In this PR, 
1. Authentication is added.
i. Login.
ii. Signup.
iii. logout.
2. Notes features have been added.
i. Create note.
ii. delete note.
iii. pin note.

please review and provide feedback.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -6,7 +6,7 @@ import { SignUp } from "./pages/authentication/signup/signup";
 import { Home } from "./pages/home/home";
 import { LandingPage } from "./pages/landingPage/landingPage";
 import { Trash } from "./pages/trash/trash";
-
```
Comment (SaurabhChawda) [inline src/App.js:9]: Create an index File for import and export.

**Review by SaurabhChawda — APPROVED** [review]
Desing and code looks good to me.

File: src/App.js [hunk 2]
```diff
@@ -16,14 +16,15 @@ function App() {
       {/* <Trash />  */}
```
Comment (tusharkandpal) [inline src/App.js:16]: Can remove these pages from here

File: src/components/textEditor/textEditor.jsx [hunk 3]
```diff
@@ -1,45 +1,84 @@
...
 import { IoColorPaletteOutline, IoAddCircleOutline, IoArchiveOutline, IoTrashOutline } from "react-icons/io5";
 import { VscCircleLargeFilled } from "react-icons/vsc";
+import { useState } from 'react';
+import { useNotes } from '../../context/notesContext/notesContext';
+
+const initialNotedata = {
+    color: '',
+    editorContent: '',
+    priorityDetails: '',
+    tags: '',
```
Comment (tusharkandpal) [inline src/components/textEditor/textEditor.jsx:11]: Tags should be an array imo.

**Review by tusharkandpal — APPROVED** [review]
Good folder structure & clean code.

Can use barrel exports & implement inline style using classes.

File: src/App.js (hunk 2 above)
Comment (ShreyashMahajan) [reply src/App.js:16]: ok

File: src/components/textEditor/textEditor.jsx (hunk 3 above)
Comment (ShreyashMahajan) [reply src/components/textEditor/textEditor.jsx:11]: ok thank you i will change that

**Pull request merged**